bravado/http_future.py

```diff
@@ -101,26 +109,51 @@ def __init__(self, future, response_adapter, operation=None,
         self.response_callbacks = response_callbacks or REQUEST_OPTIONS_DEFAULTS['response_callbacks']
         self.also_return_response = also_return_response
 
-    @reraise_errors
-    def result(self, timeout=None):
+    def response(self, timeout=None, fallback_response=None, exceptions_to_catch=FALLBACK_EXCEPTIONS):
         """Blocking call to wait for the HTTP response.
 
+        :param timeout: Number of seconds to wait for a response. Defaults to
+            None which means wait indefinitely.
+        :type timeout: float
+        :param fallback_response: callable that accepts an exception as argument and returns the
+            swagger result to use in case of errors
+        :type fallback_response: callable that takes an exception and returns a fallback swagger result
+        :param exceptions_to_catch: Exception classes to catch and call `fallback_response`
+            with. Has no effect if `fallback_response` is not provided. By default, `fallback_response`
+            will be called for read timeout and server errors (HTTP 5XX).
+        :type exceptions_to_catch: List/Tuple of Exception classes.
+        :return: A BravadoResponse instance containing the swagger result and response metadata.
+        """
+        incoming_response = None
+        try:
+            swagger_result, incoming_response = self._get_result_and_incoming_response(timeout)
```

Comment:
I would suggest to split ``incoming_response`` and ``swagger_result`` evaluation.

Let's assume that the service returns HTTP/503 from the request (the exception will be part of ``exceptions_to_catch``).
If you're extracting both objects together you'll not be able to keep trace of ``incoming_response`` as the evaluation of ``swagger_result`` calls ``unmarshal_response`` that will raise the expected bravado exception.
So this method call will return a ``BravadoResponse`` with no information about the incoming response.

My suggestion would be to split them
```python
try:
    incoming_response = self._get_incoming_response(timeout=timeout)
    swagger_result = xxx  # If this raises incoming_response has already be assigned and will be available
    ....
except exceptions_to_catch as e:
    ... 
```